### PR TITLE
MAINT: LinAlgError from public numpy.linalg

### LINKED ISSUE #14754: MAINT: `LinAlgError` not an attribute of `numpy.linalg.linalg`
### Describe your issue.

Nightly CI is [failing](https://github.com/scipy/scipy/runs/3649894246) with

```
scipy/optimize/_linprog_rs.py:26: error: Module "numpy.linalg.linalg" has no attribute "LinAlgError"  [attr-defined]
```
I believe this is due to recent changes from https://github.com/numpy/numpy/pull/19887

### PR DESCRIPTION
**Pull request by tupui**
#### Reference issue
Closes gh-14754

#### What does this implement/fix?
Don't use private members from NumPy: `numpy.linalg.linalg`.

### CODE REVIEW

File: scipy/optimize/_linprog_rs.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 from ._bglu_dense import LU
 from ._bglu_dense import BGLU as BGLU
 from scipy.linalg import LinAlgError
-from numpy.linalg.linalg import LinAlgError as LinAlgError2
+from numpy.linalg import LinAlgError as LinAlgError2
```
Comment (ev-br) [inline scipy/optimize/_linprog_rs.py:26]: This looks quite weird. What's the point of importing the same thing twice under different names. I actually checked the file, and the only other use of this seems to be this line (https://github.com/scipy/scipy/pull/14756/files#diff-7be824e870d609afd212fb11b3ee5858537ac96eaa0afe0c447a1f3998315e6eL90):

        except (LinAlgError, LinAlgError2):

Unless there is some other use of it, a better fix would be to clean up the duplicate.

Comment (ilayn) [reply scipy/optimize/_linprog_rs.py:26]: SciPy LinalgError is NumPy's LinAlgError anyways. 

https://github.com/scipy/scipy/blob/master/scipy/linalg/misc.py#L1-L6

Comment (tupui) [reply scipy/optimize/_linprog_rs.py:26]: ok then I will clean this! Thank you both.

Comment (tupui) [reply scipy/optimize/_linprog_rs.py:26]: Shall I actually move all usage of `scipy.linalg. LinAlgError` to `numpy.linalg. LinAlgError`?

Comment (tupui) [reply scipy/optimize/_linprog_rs.py]: If not, the PR is ready on my side.

**Comment by ev-br** [comment]
Let's merge it as is to keep the ball rolling , and iterate in follow-up PR. Thanks @tupui , @ilayn

**Pull request merged**